tests/functional/test_change_password.py

```diff
@@ -0,0 +1,91 @@
+import os
+import jwt
+from tests.factories import user
+
+
+class TestChangePassword(object):
+    """A User can change password"""
+
+    def test_change_password_with_correct_credentials(self, client, current_user):
+        """Post to /change-password url with correct credentials succeeds"""
+
+        res = client.post(
+            "/change-password",
+            json={
+                "email": current_user.email,
+                "old_password": "123123",
+                "new_password": "abc123",
+            },
+        )
+
+        assert res.status_code == 204
+
+    def test_change_password_and_login(self, client, current_user):
+        """Post to /change-password url and then log in successfully"""
+
+        res = client.post(
+            "/change-password",
+            json={
+                "email": current_user.email,
+                "old_password": "123123",
+                "new_password": "abc123",
+            },
+        )
+
+        assert res.status_code == 204
+
+        res = client.post(
+            "/login", json={"email": current_user.email, "password": "abc123"}
+        )
+
+        json_response = res.get_json()
+
+        assert res.status_code == 200
+
+    def test_change_password_with_incorrect_password(self, client, current_user):
+        """Post to /change-password url with incorrect password fails"""
+
+        res = client.post(
+            "/change-password",
+            json={
+                "email": current_user.email,
+                "old_password": "definitely-wrong",
+                "new_password": "abc123",
+            },
+        )
+
+        assert res.status_code == 403
+
+    def test_change_password_with_wrong_email(self, client, current_user):
+        """Post to /change-password url with wrong email fails"""
+
+        res = client.post(
+            "/change-password",
+            json={
+                "email": "jane@wehadababyitsagirl.com",
+                "old_password": "123123",
+                "new_password": "abcabc",
+            },
+        )
+
+        assert res.status_code == 403
```

Comment:
We should return 200 for this - otherwise it exposes users emails to harvesting.